flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt

```diff
@@ -6,94 +6,163 @@
+      val inspector = flutterInspectorFixture(this)
+      inspector.populate()
+      val widgets = inspector.widgetsFixture()
+      val widgetsTree = widgets.inspectorTreeFixture(isDetails = false)
+      widgetsTree.selectRow(0)
+      val detailsTree = widgets.inspectorTreeFixture(isDetails = true)
+      val initialDetails = detailsTree.selectionSync().toString()
+
+      editor {
+        // Wait until current file has appeared in current editor and set focus to editor.
+        moveTo(0)
+        val editorCode = getCurrentFileContents(false)!!
+        val original = "pushed the button this"
+        val index = editorCode.indexOf(original) + original.length
+        moveTo(index)
+        val key = if (isMac) KeyEvent.VK_BACK_SPACE else KeyEvent.VK_DELETE
+        for (n in 1..4) typeKey(key)
+        typeText("that")
+        //typeKey(KeyEvent.VK_ESCAPE) // Dismiss completion popup -- not needed with "that" but is needed with "so"
+      }
+
+      step("Trigger Hot Reload and wait for it to finish") {
+        val reload = findHotReloadButton()
+        reload.click()
+        editor.clickCenter() // Need to cycle the event loop to get button enabled on Mac.
+        pause(object : Condition("Hot Reload finishes") {
+          override fun test(): Boolean {
+            return reload.isEnabled
+          }
+        }, Timeouts.seconds05)
+        step("Work around #3370") {
+          // https://github.com/flutter/flutter-intellij/issues/3370
+          inspector.renderTreeFixture().show() // The refresh button is broken so force tree update by switching views.
+          inspector.populate()
+          widgets.show() // And back to the one we want
+        }
+        widgetsTree.selectRow(6) // Text widget
+        pause(object : Condition("Details tree changes") {
+          override fun test(): Boolean {
+            return initialDetails != detailsTree.selectionSync().toString()
+          }
+        }, Timeouts.seconds05)
+        val row: JTreeRowFixture = detailsTree.treeFixture().node(1)
+        val expected = "\"You have pushed the button that many times:\""
+        expect(expected) { row.value() }
+      }
+
+      runner().stop()
+    }
   }
 
   fun IdeFrameFixture.launchFlutterApp() {
-    findRunApplicationButton().click()
-    val runner = runner()
-    pause(object : Condition("Start app") {
-      override fun test(): Boolean {
-        return runner.isExecutionInProgress
-      }
-    }, Timeouts.seconds10)
+    step("Launch Flutter app") {
+      findRunApplicationButton().click()
+      val runner = runner()
+      pause(object : Condition("Start app") {
+        override fun test(): Boolean {
+          return runner.isExecutionInProgress
+        }
+      }, Timeouts.seconds30)
+    }
   }
 
-  fun IdeFrameFixture.selectSimulator() {
-    selectDevice("Open iOS Simulator")
+  private fun findHotReloadButton(): ActionButtonFixture {
+    return findActionButtonByClassName("ReloadFlutterAppRetarget")
```

Comment:
To make this slightly more robust could we consider something like `ReloadFlutterAppRetarget.class.getName()` (or whatever it is in Java)?

Reply:
Great idea. I'll look into that. Next PR?

Reply:
Great!